animation_nodes/nodes/gpencil/replicate_gp_layer.py

```diff
@@ -0,0 +1,80 @@
+import bpy
+from bpy.props import *
+from mathutils import Matrix
+from ... data_structures import GPLayer, VirtualDoubleList
+from ... base_types import AnimationNode, VectorizedSocket
+
+transformationTypeItems = [
+    ("Matrix List", "Matrices", "", "NONE", 0),
+    ("Vector List", "Vectors", "", "NONE", 1)
+]
+
+class ReplicateGPLayerNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_ReplicateGPLayerNode"
+    bl_label = "Replicate GP Layer"
+
+    useLayerList: VectorizedSocket.newProperty()
+
+    transformationType: EnumProperty(name = "Transformation Type", default = "Matrix List",
+        items = transformationTypeItems, update = AnimationNode.refresh)
+
+    def create(self):
+        self.newInput(VectorizedSocket("GPLayer", "useLayerList",
+            ("Layer", "layer", dict(defaultDrawType = "PROPERTY_ONLY")),
+            ("Layers", "layers")))
+        self.newInput(self.transformationType, "Transformations", "transformations")
+        self.newInput("Float List", "Offset Frame Numbers", "offsets", value = 0, hide = True)
+        self.newInput("Boolean", "Name Suffix", "nameSuffix", value = False, hide = True)
```

Comment:
I think we should force the suffix. So no need for this input.

Reply:
Okay.